src/main/java/uk/gov/companieshouse/model/validator/CompanyNumberValidator.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.companieshouse.model.validator;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.jboss.logging.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CompanyNumberValidator implements ConstraintValidator<ValidCompanyNumber, String> {
+
+    private final Pattern companyNumberRegex;
+
+    public CompanyNumberValidator(@Value("${companyNumber.prefixes}") String prefixString) {
+
+        Pattern prefixListRegex = Pattern.compile("(?i)^([A-Z][A-Z]?)\\b(,[A-Z][A-Z]?)*$");
+
+        if (!prefixListRegex.matcher(prefixString).matches()) {
```

Comment:
if `prefixString` is null, the regex matcher will throw a null pointer exception, we should ensure this doesn't happen

Reply:
![image](https://user-images.githubusercontent.com/54070318/85582996-ca31ba80-b635-11ea-81a8-dd9478d4814b.png)